stolon 0.13.0 (new formula)

dineshba opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

SMillerDev commented on Formula/stolon.rb:
```diff
@@ -0,0 +1,33 @@
+$version = "v0.13.0"
```
these definitions shouldn't be here.

dineshba replied:
We removed it.

SMillerDev commented on Formula/stolon.rb:
```diff
@@ -0,0 +1,33 @@
+    desc "Stolon is a cloud native PostgreSQL manager for high availability"
+    homepage "https://github.com/sorintlab/stolon"
+    url "https://github.com/sorintlab/#{$bin_name}.git", tag: $version
+    version $version
```
this isn't needed, the version should be picked up from the tag

dineshba replied:
Removed the version

SMillerDev commented on Formula/stolon.rb:
```diff
@@ -0,0 +1,33 @@
+    url "https://github.com/sorintlab/#{$bin_name}.git", tag: $version
+    version $version
+
+    depends_on "go" => :recommended
```
Don't use recommended. It doesn't build without go.

dineshba replied:
we made it `build`

SMillerDev commented on Formula/stolon.rb:
```diff
@@ -0,0 +1,33 @@
+    version $version
+
+    depends_on "go" => :recommended
+    depends_on "postgresql" => :recommended
```
This is a runtime requirement as far as I can tell, that means it's not just recommended

dineshba replied:
As `postgresql` is needed for stolon at runtime, we removed `recommended`

SMillerDev commented on Formula/stolon.rb:
```diff
@@ -0,0 +1,33 @@
+
+    def install
+        group_dir = "src/github.com/sorintlab"
+        source_dir = "#{group_dir}/#{$bin_name}"
```
please just use a inline string here

dineshba replied:
Changed

SMillerDev commented on Formula/stolon.rb:
```diff
@@ -0,0 +1,33 @@
+        bin.install "./bin/stolon-sentinel" => "stolon-sentinel"
+        bin.install "./bin/stolonctl" => "stolonctl"
+        end
+    end
```
This formula is missing a meaningful test, please provide one that tests some functionality.

dineshba replied:
We tested the version check

SMillerDev commented on Formula/stolon.rb:
```diff
@@ -0,0 +1,33 @@
+
+        Dir.chdir buildpath/source_dir do
+        system "./build"
+        bin.install "./bin/stolon-keeper" => "stolon-keeper"
```
the arguments here seem unneeded as that's already the filename.

dineshba replied:
Thanks. We changed accordingly.

dineshba commented:
Thanks @SMillerDev for the feedbacks. Will fix all those things and get back to you.

dineshba commented:
Hi @SMillerDev. Thanks a lot for the feedback. We have changed/refactored a lot. We added test and ran the audit command in our local. Please have a look and provide your inputs.

fxcoudert commented on Formula/stolon.rb:
```diff
@@ -0,0 +1,30 @@
+    run_output = shell_output("#{bin}/stolon-sentinel --version 2>&1")
+    assert_match "stolon-sentinel version v0.13.0", run_output
+    run_output = shell_output("#{bin}/stolon-proxy --version 2>&1")
+    assert_match "stolon-proxy version v0.13.0", run_output
```
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

dineshba replied:
Hi @fxcoudert Thanks for your inputs. I added test for `stolonctl`(refer my commit). I am still figuring out how to write tests for other three (stolon-keeper,stolon-sentinel,stolon-proxy) as they are long running processes. Is that okay if check only the version for long running processes? Can you please provide some direction. Thanks.

fxcoudert replied:
I think having a full test for one of them is probably enough, thanks

dineshba replied:
Thanks for your inputs. I squashed the commits and did force push to have clean history.

fxcoudert commented:
Looks good to me, @SMillerDev any more comments?

SMillerDev commented:
Looks good to me too. Good job @dineshba

dineshba commented:
Thanks @fxcoudert  and @SMillerDev.  @SMillerDev Can I resolve the comments?
